### PR TITLE
ephemeral pr envs on label

### PR DESCRIPTION
**Pull request by oreonl**
### Description

Adds the ability to spin up a PR Env by adding the label `PR-Env`.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Label the PR with `PR-Env`
2. Ensure PR Env comes up and is accessible

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -251,3 +251,11 @@ jobs:
...
       - run: |
           echo "Didn't run due to conditional filtering"
+
+  pr-env:
+    needs: [build]
+    if: |
+      !cancelled() &&
+      contains(github.event.pull_request.labels.*.name, 'PR-Env')
+    uses: ./.github/workflows/pr-env.yml
+    secrets: inherit
```
Comment (nemanjaglumac) [inline .github/workflows/e2e-tests.yml:261]: Why do we do this in E2E workflow?
Isn't https://github.com/metabase/metabase/blob/master/.github/workflows/uberjar.yml a better choice?

Comment (oreonl) [reply .github/workflows/e2e-tests.yml:261]: So that we can use the uberjar artifacts that are built as part of the E2E workflow.

Comment (nemanjaglumac) [reply .github/workflows/e2e-tests.yml:261]: Oh, I forgot that the uberjar.yml is building only on push to `master`.

**Review by iethree — APPROVED** [review]
looks good to me!

**Comment by oreonl** [comment]
Notion page https://www.notion.so/metabase/Ephemeral-PR-Environments-c74a3710c87a460dbf4fc0007a001458

**Pull request merged**